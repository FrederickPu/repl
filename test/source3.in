{"cmd": "import Lean\ndef f := 1\ndef g := 2\nexample : f + 1 = g := rfl"}

{"cmd": "def f := 3", "replace": 1}

{"src": 4}

{"src": 5}

{"src": 4, "after": true}

{"cmd": "def g := 4", "replace": 5}

{"src": 0, "after": true}

