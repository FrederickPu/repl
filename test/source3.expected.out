{"results":
 [{"source": "import Lean\n  \n  ", "env": 0},
  {"source": "def f :=\n  1", "env": 1},
  {"source": "def g :=\n  2", "env": 2},
  {"source": "example : f + 1 = g :=\n  rfl", "env": 3}]}

{"results":
 [{"source": "def f :=\n  3", "env": 4},
  {"source": "def g :=\n  2", "env": 5},
  {"source": "example : f + 1 = g :=\n  rfl",
   "messages":
   [{"severity": "error",
     "pos": {"line": 6, "column": 2},
     "endPos": {"line": 6, "column": 5},
     "data":
     "type mismatch\n  rfl\nhas type\n  f + 1 = f + 1 : Prop\nbut is expected to have type\n  f + 1 = g : Prop"}],
   "env": 6}]}

{"src": "import Lean\n  \n  \n\ndef f :=\n  3"}

{"src": "import Lean\n  \n  \n\ndef f :=\n  3\n\ndef g :=\n  2"}

{"src":
 "import Lean\n  \n  \n\ndef f :=\n  3\n\n\n\ndef g :=\n  2\n\nexample : f + 1 = g :=\n  rfl"}

{"results":
 [{"source": "def g :=\n  4", "env": 7},
  {"source": "example : f + 1 = g :=\n  rfl", "env": 8}]}

{"src":
 "import Lean\n  \n  \n\n\n\ndef f :=\n  3\n\ndef g :=\n  4\n\nexample : f + 1 = g :=\n  rfl"}

